### PR TITLE
bfs, dfs, & connected components done. Will keep trying implementing …

### PR DESCRIPTION
**Pull request by coleferg**
…user selection & animation.

### CODE REVIEW

File: projects/graph/src/App.js [hunk 1]
```diff
@@ -28,25 +62,118 @@ class GraphView extends Component {
...
   
+  handleButton() {
+    let newGraph = new Graph()
+    newGraph.randomize(5, 4, 150, 0.6)
+    this.setState({
+      graph: newGraph
+    })
+  }
+
+  handleMouse(e) {
```
Comment (coleferg) [inline projects/graph/src/App.js]: Although this function is incomplete, I was able to find the vertex that is highlighted.

File: projects/graph/src/App.js [hunk 2]
```diff
@@ -3,13 +3,47 @@ import { Graph } from './graph';
 import './App.css';
 
 // !!! IMPLEMENT ME
-// const canvasWidth = 
-// const canvasHeight = 
+const canvasWidth = 750
+const canvasHeight = 600
+
+const invertColor = (hex) => {
```
Comment (coleferg) [inline projects/graph/src/App.js]: Found on S/O so random colors would be complementary

File: projects/graph/src/App.js [hunk 3]
```diff
@@ -28,25 +62,118 @@ class GraphView extends Component {
...
 
-    // !!! IMPLEMENT ME
-    // compute connected components
-    // draw edges
-    // draw verts
-    // draw vert values (labels)
+    const connectedGraph = graph.getConnectedComponents()
+    ctx.lineWidth = 2;
+    for (let i = 0; i < connectedGraph.length; i++) {
+      const getRandomColor = () => {
```
Comment (coleferg) [inline projects/graph/src/App.js]: Random color function found on S/O. On review, I think I could have figured this out myself.

**Comment by beejjorgensen** [comment]
Good job on DFS and CC. Edge weights, graph looks super nice. And looks like the start of some mouse-over on the nodes, too...

Well done!

**Comment by coleferg** [comment]
I actually have a newer commit that has fastest route node connection from
mouse click! It was a fun project.

**Pull request closed without merging**